services/app-web/src/pages/StateSubmission/ContractDetails/ContractDetailsSchema.ts

```diff
@@ -1,82 +1,83 @@
-                        'contractDateStart',
-                        (contractDateStart: Date, schema: Yup.DateSchema) => {
-                            const startDate = dayjs(contractDateStart)
-                            if (startDate.isValid()) {
-                                return schema.min(
-                                    startDate.add(1, 'day'),
-                                    'The end date must come after the start date'
-                                )
-                            }
-                        }
-                    )
-            )
-        }
-    }),
-    managedCareEntities: Yup.array().when('contractType', {
-        is: (contractType: string | undefined) => contractType,
-        then: Yup.array().min(1, 'You must select at least one entity'),
-    }),
-    federalAuthorities: Yup.array().when('contractType', {
-        is: (contractType: string | undefined) => contractType,
-        then: Yup.array().min(1, 'You must select at least one authority'),
-    }),
+export const ContractDetailsFormSchema = (contractType: ContractType) =>
+    Yup.object().shape({
+        contractExecutionStatus: Yup.string().defined(
+            'You must select a contract status'
+        ),
+        contractDateStart: Yup.date()
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore-next-line
+            .validateDateFormat('YYYY-MM-DD', true)
+            .typeError('The start date must be in MM/DD/YYYY format')
+            .defined('You must enter a start date'),
+
+        contractDateEnd: Yup.date()
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore-next-line
+            .validateDateFormat('YYYY-MM-DD', true)
+            .typeError('The end date must be in MM/DD/YYYY format')
+            .defined('You must enter an end date')
+            .when(
+                // ContractDateEnd must be at minimum the day after Start
+                'contractDateStart',
+                (contractDateStart: Date, schema: Yup.DateSchema) => {
+                    const startDate = dayjs(contractDateStart)
+                    if (startDate.isValid()) {
+                        return schema.min(
+                            startDate.add(1, 'day'),
+                            'The end date must come after the start date'
+                        )
+                    }
+                }
+            ),
+
+        managedCareEntities: Yup.array().test(
+            'entitySelection',
+            'You must select at least one entity',
+            (value) => {
+                return Boolean(value && value.length > 0)
+            }
+        ),
+        federalAuthorities: Yup.array().test(
+            'authoritySelection',
+            'You must select at least one authority',
+            (value) => {
+                return Boolean(value && value.length > 0)
+            }
+        ),
```

Comment:
There might be a way to put all of these in a conditional inside this yup object instead of making the yesNoError take a parameter